src/main/java/net/minecraftforge/fml/client/GuiModList.java

```diff
@@ -441,38 +440,52 @@ else if (selectedMod.getVersion().equals(selectedMod.getDisplayVersion()))
-                lines.add("Update Available: " + (vercheck.url == null ? "" : vercheck.url));
-
-            lines.add(null);
-            lines.add(selectedMod.getMetadata().description);
         }
         else
         {
             lines.add(WHITE + selectedMod.getName());
             lines.add(WHITE + "Version: " + selectedMod.getVersion());
             lines.add(WHITE + "Mod State: " + Loader.instance().getModState(selectedMod));
-            if (vercheck.status == Status.OUTDATED || vercheck.status == Status.BETA_OUTDATED)
-                lines.add("Update Available: " + (vercheck.url == null ? "" : vercheck.url));
-
-            lines.add(null);
-            lines.add(RED + "No mod information found");
-            lines.add(RED + "Ask your mod author to provide a mod mcmod.info file");
         }
 
-        if ((vercheck.status == Status.OUTDATED || vercheck.status == Status.BETA_OUTDATED) && vercheck.changes.size() > 0)
+        CheckResult vercheck = ForgeVersion.getCleanResult(selectedMod);
+        if (vercheck != null && (vercheck.status == Status.OUTDATED || vercheck.status == Status.BETA_OUTDATED || vercheck.status == Status.BETA || vercheck.status == Status.AHEAD))
         {
             lines.add(null);
-            lines.add("Changes:");
-            for (Entry<ComparableVersion, String> entry : vercheck.changes.entrySet())
+            if (vercheck.status == Status.BETA)
+                lines.add(GOLD + "This version is a beta version");
+            else if (vercheck.status == Status.AHEAD)
+                lines.add(LIGHT_PURPLE + "This version is ahead of the latest version found (" + LIGHT_PURPLE + vercheck.latestFound + ")");
+            else if (vercheck.status == Status.BETA_OUTDATED)
+                lines.add(GOLD + "Beta update (" + GOLD + vercheck.latestFound + ") available: " + (vercheck.homepage == null ? "" : vercheck.homepage));
+            else
+                lines.add(GREEN + "Update (" + GREEN + vercheck.latestFound + ") available: " + (vercheck.homepage == null ? "" : vercheck.homepage));
+
+            if (!vercheck.changes.isEmpty())
             {
-                lines.add("  " + entry.getKey() + ":");
-                lines.add(entry.getValue());
                 lines.add(null);
+                lines.add("Changes:");
+                for (Entry<ComparableVersion, String> entry : vercheck.changes.entrySet())
+                {
+                    lines.add("  " + entry.getKey() + ":");
+                    lines.add(entry.getValue());
+                    lines.add(null);
+                }
             }
         }
 
+        if (!selectedMod.getMetadata().autogenerated)
+        {
+            lines.add(null);
+            lines.add(selectedMod.getMetadata().description);
+        }
+        else
+        {
+            lines.add(null);
+            lines.add(RED + "No mod information found");
+            lines.add(RED + "Ask your mod author to provide a mod mcmod.info file");
```

Comment:
`Ask your mod author to provide a mod mcmod.info file` => `Ask the mod author to provide a mcmod.info file`